### PR TITLE
TST: Parametrize tests in tools/test_numeric.py

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.

### CODE REVIEW

File: pandas/tests/tools/test_numeric.py [hunk 1]
```diff
@@ -227,17 +232,17 @@ def test_str(self):
         res = pd.to_numeric(idx.values)
         tm.assert_numpy_array_equal(res, exp)
 
-    def test_datetimelike(self):
-        for tz in [None, 'US/Eastern', 'Asia/Tokyo']:
-            idx = pd.date_range('20130101', periods=3, tz=tz, name='xxx')
-            res = pd.to_numeric(idx)
-            tm.assert_index_equal(res, pd.Index(idx.asi8, name='xxx'))
+    @pytest.mark.parametrize("tz", [None, "US/Eastern", "Asia/Tokyo"])
```
Comment (jschendel) [inline pandas/tests/tools/test_numeric.py]: Should the common `tz_naive_fixture` be used here?  It has a few extra timezones, but I'm not sure if it tests anything extra/if there's a benefit to including the extra ones:

https://github.com/pandas-dev/pandas/blob/7cd2679982138e1e7f9f1a759827647e6e521d43/pandas/conftest.py#L221-L230

Comment (jreback) [reply pandas/tests/tools/test_numeric.py]: yes ideally we re-use our fixtures as much as possible

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/tools/test_numeric.py (hunk 1 above)
Comment (gfyoung) [reply pandas/tests/tools/test_numeric.py]: That makes sense to me.  Done.

**Comment by gfyoung** [comment]
@jreback @jschendel : Comments addressed, and all is still green.  PTAL.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @gfyoung very nice!

**Pull request merged**